src/lib/yang/schema.lua

```diff
@@ -794,14 +794,15 @@ function resolve(schema, features)
          end
       end
       if node.body then
-         node.body = shallow_copy(node.body)
-         for k,v in pairs(node.body or {}) do
+         local body = node.body or {}
+         node.body = {}
+         for k,v in pairs(body) do
             if v.kind == 'uses' then
                -- Inline "grouping" into "uses".
                local grouping = lookup_lazy(env, 'groupings', v.id)
-               node.body[k] = nil
+               body[k] = nil
                for k,v in pairs(grouping.body) do
-                  assert(not node.body[k], 'duplicate identifier: '..k)
+                  assert(not body[k], 'duplicate identifier: '..k)
                   node.body[k] = v
                end
                -- TODO: Handle refine and augment statements.
```

Comment:
can we add an `assert(not node.body[k])` in the general case as well please on https://github.com/Igalia/snabb/pull/831/files#diff-930fd503df3ea0198278a8c70917708fR810.